lib/internal/http2/core.js

```diff
@@ -2178,7 +2181,7 @@ class ServerHttp2Stream extends Http2Stream {
     let headRequest = false;
     if (headers[HTTP2_HEADER_METHOD] === HTTP2_METHOD_HEAD)
       headRequest = options.endStream = true;
-    options.readable = !options.endStream;
+    options.readable = false;
```

Comment:
@jasnell I remember we discussed this, and you had a very good reason why this was there. Can you please weight in?

Reply:
If I am recalling correctly, this was there because of an earlier iteration. I don't think this is necessary now.